src/utils/enum_helper.h

```diff
@@ -66,6 +66,11 @@
 
 #define ENUM_END(type) ENUM_END2(type, type)
 
+#define ENUM_CONST(str) k##str
+#define ENUM_CONST_DEF(str) ENUM_CONST(str),
+#define ENUM_CONST_REG_STR(enum_class, str)                                                        \
+    helper->register_enum(#str, enum_class::ENUM_CONST(str));
```

Comment:
It would be great if adding a sample of how to use the serial of macros, and the related enum helpers, e.g. convert from/to std::string.

Reply:
OK, I'll add some sample comments to describe how to use these macros.